iocore/net/QUICNetVConnection.cc

```diff
@@ -1131,12 +1131,33 @@ QUICNetVConnection::_store_frame(ats_unique_buf &buf, size_t &len, bool &retrans
 
   // TODO: check debug build
   char msg[1024];
-  frame->debug_msg(msg, sizeof(msg));
   QUICConDebug("[TX] %s", msg);
 
-  frame->store(buf.get() + len, &l);
-  len += l;
+  ink_assert(max_size > len);
+  size_t n = frame->store(buf.get() + len, &l, max_size - len);
+  if (n > 0) {
+    frame->debug_msg(msg, sizeof(msg));
+    len += l;
+    return;
+  }
+
+  // split frame
+  auto new_frame = QUICFrameFactory::split_frame(frame.get(), max_size - len);
+  if (new_frame != nullptr) {
+    frame->debug_msg(msg, sizeof(msg));
+    ink_assert(frame->store(buf.get() + len, &l, max_size - len) > 0);
+    this->_transmit_packet(this->_build_packet(std::move(buf), len, retransmittable, current_packet_type));
+    len = 0;
+    buf = ats_unique_malloc(max_size);
+    this->_store_frame(buf, len, retransmittable, current_packet_type, std::move(new_frame));
+    return;
+  }
 
+  // failed to split frame, send the previous one and create new buf to hold the large frame.
```

Comment:
I don't think we need this block. This case should be covered by line 1117-1128 because `split()` can fail only if the specified size is smaller than frame header size and minimum packet size should always be bigger than that.

Reply:
Currently the MAX_PACKET_OVERHEAD is always larger than Stream Frame's header size, so it works fine. 

But it will assert when MAX_PACKET_OVERHEAD is smaller than Stream Frame's header size(May never happen).